Search/scrape_plugins.py

```diff
@@ -142,24 +145,38 @@ def overlayForEngine(engine):
         return None
     return Overlay(path)
 
-def writeList(engines):
+def saveSearchEngines(engines, searchEngineDefaults):
     root = etree.Element('dict')
     for locale in sorted(engines.keys()):
         key = etree.Element('key')
         key.text = locale
         root.append(key)
-        values = etree.Element('array')
-        for engine in engines[locale]:
-            value = etree.Element('string')
-            value.text = engine
-            values.append(value)
-        root.append(values)
 
+        localeDict = etree.Element('dict')
+        searchEngineDefaultKey = etree.Element('key')
+        searchEngineDefaultKey.text = 'searchEngineDefault'
+        localeDict.append(searchEngineDefaultKey)
+
+        searchEngineDefaultValue = etree.Element('string')
+        searchEngineDefaultValue.text = searchEngineDefaults[locale]
+        localeDict.append(searchEngineDefaultValue)
+
+        visibleEnginesKey = etree.Element('key')
+        visibleEnginesKey.text = 'visibleEngines'
+        localeDict.append(visibleEnginesKey)
+
+        visibleEnginesValue = etree.Element('array')
+        for engine in engines[locale]:
+            visibleEngine = etree.Element('string')
+            visibleEngine.text = engine
+            visibleEnginesValue.append(visibleEngine)
+        localeDict.append(visibleEnginesValue)
+        root.append(localeDict)
+    
     plist = etree.tostring(root, encoding="utf-8", pretty_print=True)
     with open("SearchEngines.plist", "w") as outfile:
         outfile.write(plist)
 
-
```

Comment:
nit: whitespace change. I know this *shouldn't* be here, but if we do a git blame it'll point to you, so let's keep it there for now.

Whitespace cleanup should be a separate issue we tackle--in fact I've proposed a linting system (#1430) that hopefully gets implemented soon that will take care of this.